### PR TITLE
Added clear button to Media search pane

### LINKED ISSUE #701: Media Panel - Search Bar Missing X to Clear Query
## Bug Description

After entering a search term there is no X icon to clear the search field.

image

Taken from Claudio in #585

## Screenshots

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by winterDroid**
## Summary

Adds a clear button according to [Figma](https://www.figma.com/file/Cd9iacVknHK62RO4pRwb68/Stories-Editor?node-id=8617%3A46576).

| Before | After |
| ------------- | ------------- |
| Screen Shot 2020-06-05 at 11 30 04 am  | Screen Shot 2020-06-05 at 11 28 57 am  |
| Screen Shot 2020-06-05 at 11 30 14 am  | Screen Shot 2020-06-05 at 11 29 11 am  |

## Relevant Technical Choices

I reused the existing `TextInput` and added some more customization options.

Fixes #701

### CODE REVIEW

File: assets/src/edit-story/components/library/common/searchInput.js [hunk 1]
```diff
@@ -19,7 +19,11 @@
  */
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
-import { rgba } from 'polished';
+/**
```
Comment (cvolzke4) [inline assets/src/edit-story/components/library/common/searchInput.js:22]: Internal dependencies should be placed after WordPress dependencies.

File: assets/src/edit-story/components/form/text.js [hunk 2]
```diff
@@ -89,13 +88,27 @@ function TextInput({
   flexBasis,
   disabled,
   clear,
```
Comment (cvolzke4) [inline assets/src/edit-story/components/form/text.js:90]: Not related to these changes, but I found the name 'clear' not easily parsable, perhaps this should be named 'isClearIconVisible'.

**Comment by cvolzke4** [comment]
Is it possible to add a before and after screenshot?

**Review by cvolzke4 — APPROVED** [review]

**Comment by winterDroid** [comment]
> Is it possible to add a before and after screenshot?

Done.

File: assets/src/edit-story/components/form/text.js (hunk 2 above)
Comment (winterDroid) [reply assets/src/edit-story/components/form/text.js:90]: I agree. But as this is used in other places as well, I'll keep it like that.

**Pull request merged**